keychain_linux.cpp

```diff
@@ -89,21 +104,18 @@ std::string getPassword(const std::string &package, const std::string &service,
                                                 NULL);
 
     std::string password;
+
     if (error != NULL) {
-        err.error = KeychainError::GenericError;
-        err.message = error->message;
-        err.code = error->code;
-        g_error_free(error);
-        return "";
+        handleError(error, err);
     } else if (raw_passwords == NULL) {
         err.error = KeychainError::NotFound;
         err.message = "Password not found.";
         err.code = -1; // generic non-zero
-        return "";
     } else {
         password = raw_passwords;
+        secret_password_free(raw_passwords);
     }
-    secret_password_free(raw_passwords);
+
     return password;
 }
```

Comment:
By making the function always return the local variable `password` we should benefit from RVO.